client/sources/ok_test/models.py

```diff
@@ -163,13 +165,22 @@ def dump(self):
         # directory may be left in a corrupted state.
         # TODO(albert): might need to delete obsolete test files too.
         json = format.prettyjson(self.to_json())
-        with open(self.file, 'w', encoding='utf-8') as f:
+        test_tmp = "{}.tmp".format(self.file)
+
+        # Use an atomic rename operation to prevent test corruption
+
+        with open(test_tmp, 'w', encoding='utf-8') as f:
             f.write('test = ' + json)
```

Comment:
Any reason to not have a new line character at the end? 

             f.write('test = ' + json + '\n')